tests/lax_test.py

```diff
@@ -1714,26 +1714,42 @@ def testTransposeAgainstNumpy(self, shape, dtype, perm):
       {"testcase_name": "_op={}_inshape={}_reducedims={}_initval={}"
        .format(op.__name__, jtu.format_shape_dtype_string(shape, dtype), dims,
                init_val),
-       "op": op, "init_val": init_val, "shape": shape, "dtype": dtype, "dims": dims}
-      for init_val, op, types in [
-          (0, lax.add, default_dtypes),
-          (1, lax.mul, default_dtypes),
-          (0, lax.max, all_dtypes), # non-monoidal
-          (-np.inf, lax.max, float_dtypes),
-          (dtypes.iinfo(np.int32).min, lax.max, [np.int32]),
-          (dtypes.iinfo(np.int64).min, lax.max, [np.int64]),
-          (np.inf, lax.min, float_dtypes),
-          (dtypes.iinfo(np.int32).max, lax.min, [np.int32]),
-          (dtypes.iinfo(np.int64).max, lax.min, [np.int64]),
-          (dtypes.iinfo(np.uint32).max, lax.min, [np.uint32]),
-          (dtypes.iinfo(np.uint64).max, lax.min, [np.uint64]),
+       "op": op, "init_val": init_val, "shape": shape, "dtype": dtype, "dims": dims, "prim": prim}
+      for init_val, op, types, prim in [
+          (0, lax.add, default_dtypes, lax.reduce_sum_p),
+          (1, lax.mul, default_dtypes, lax.reduce_prod_p),
+          (0, lax.max, float_dtypes + complex_dtypes + int_dtypes, lax.reduce_p), # non-monoidal
+          (0, lax.max, uint_dtypes + bool_dtypes, lax.reduce_max_p),
+          (-np.inf, lax.max, float_dtypes, lax.reduce_max_p),
+          (dtypes.iinfo(np.int32).min, lax.max, [np.int32], lax.reduce_max_p),
+          (dtypes.iinfo(np.int64).min, lax.max, [np.int64], lax.reduce_max_p),
+          (np.inf, lax.min, float_dtypes, lax.reduce_min_p),
+          (dtypes.iinfo(np.int32).max, lax.min, [np.int32], lax.reduce_min_p),
+          (dtypes.iinfo(np.int64).max, lax.min, [np.int64], lax.reduce_min_p),
+          (dtypes.iinfo(np.uint32).max, lax.min, [np.uint32], lax.reduce_min_p),
+          (dtypes.iinfo(np.uint64).max, lax.min, [np.uint64], lax.reduce_min_p),
+          (dtypes.iinfo(np.uint32).max, lax.bitwise_and, [np.uint32], lax.reduce_and_p),
+          (dtypes.iinfo(np.uint64).min, lax.bitwise_or, [np.uint64], lax.reduce_or_p),
+          (0, lax.bitwise_xor, uint_dtypes, lax.reduce_xor_p)
```

Comment:
This parameterization is starting to look a bit daunting. Perhaps we should define a `ReducerOpRecord` class to help organize which cases we want to test?

Reply:
Fine by me, is there a pre-existing op record implementation in JAX testing that I should use?